src/aircrack-ng.c

```diff
@@ -1391,6 +1394,28 @@ static void read_thread(void *arg)
 					break;
 				}
 			}
+			append_ap(ap_cur);
+		}
+		else
+		{
```

Comment:
The fix looks good but in 6 months, we won't remember why this code is here (that look like a duplicate of the code above), so could you add a comment with a small explanation of the issue and reference the URL where the issue is discovered (of the exact post in this PR)?